gluten-substrait/src/main/scala/org/apache/gluten/execution/WriteFilesExecTransformer.scala

```diff
@@ -127,6 +131,37 @@ case class WriteFilesExecTransformer(
 
   override protected def doValidateInternal(): ValidationResult = {
     val finalChildOutput = getFinalChildOutput
+
+    def isConstantComplexType(e: Expression): Boolean = {
+      e match {
+        case Literal(_, _: ArrayType | _: MapType) => true
```

Comment:
@weixiuli Do we need StructType check here?

Reply:
> @weixiuli Do we need StructType check here?

Whether or not a StructType is a constant, it is captured by the validateDataTypes and does not support vectorized writes. The isConstantComplexType method is used to check if ArrayType/MapType is sconstant, and if it is, it needs to fall back because velox currently supports this scenario.